File: lab3/task_5/java_fixed/untitled/src/Jewelry.java
import java.util.Scanner;

public class Jewelry {
    private double weight = 1;
    private double pricePerGram = 1;

    public void init(double weight, double pricePerGram) {
        this.weight = weight;
        this.pricePerGram = pricePerGram;
    }

    public void read() {
        Scanner sc = new Scanner(System.in);
        System.out.print("Введите вес в граммах: ");
        this.weight = sc.nextDouble();
        System.out.print("Введите цену за грамм: ");
        this.pricePerGram = sc.nextDouble();
    }

    public void display() {
        System.out.println("Вес: " + weight);
        System.out.println("Цена за грамм: " + pricePerGram);
    }

    public double getFullPricePerGram() {
        return weight * pricePerGram;
    }
}

File: part_2/lab1/patterns/Jewelry.cs
using System;

namespace lab1
{
    public class Jewelry
    {
        public double weight;
        public double pricePerGramm;

        public Jewelry()
        {
            weight = 0;
            pricePerGramm = 0;
        }

        public Jewelry(double w, double p)
        {
            weight = w;
            pricePerGramm = p;
        }

        public void Init(double arg_weight, double arg_pricePerGramm)
        {
            weight = arg_weight;
            pricePerGramm = arg_pricePerGramm;
        }

        public void Read()
        {
            Console.Write("Введите вес в граммах: ");
            weight = Convert.ToDouble(Console.ReadLine());
            Console.Write("Введите цену за грамм: ");
            pricePerGramm = Convert.ToDouble(Console.ReadLine());
        }


        public void Display()
        {
            Console.WriteLine($"Вес: {weight}");
            Console.WriteLine($"Цена за грамм: {pricePerGramm}");
        }

        public double GetFullPricePerGramm()
        {
            return weight * pricePerGramm;
        }
    }
}
